### PR TITLE
Add unit test for ImageList

### PR DESCRIPTION
**Pull request by vdemeester**
Time for `ImageList` 🐮.

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by calavera** [comment]
LGTM

File: client/image_list_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	listCases := []struct {
+		options             types.ImageListOptions
+		expectedQueryParams map[string]string
+	}{
+		{
+			options: types.ImageListOptions{},
+			expectedQueryParams: map[string]string{
+				"all":     "",
+				"filter":  "",
+				"filters": "",
```
Comment (duglin) [inline client/image_list_test.go]: don't want to test `filters`?

Comment (vdemeester) [inline client/image_list_test.go]: @duglin oh damn you are right… I wanted to but.. seems I forgot :stuck_out_tongue_closed_eyes: Will update :wink:

Comment (vdemeester) [inline client/image_list_test.go]: @duglin updated below :wink:

File: client/image_list_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			},
+		},
+		{
+			options: types.ImageListOptions{
+				Filters: filters,
+			},
+			expectedQueryParams: map[string]string{
+				"all":     "",
+				"filter":  "",
+				"filters": `{"dangling":{"true":true},"label":{"label1":true,"label2":true}}`,
```
Comment (duglin) [inline client/image_list_test.go]: that is just freaky.  Could you add a test for "dangling=true" so we can check that the json shows up as "false": true - which is also mindblowing  :-)

Comment (vdemeester) [inline client/image_list_test.go]: I can't do that yeah :wink:. And yeah, the way filters are serialized made be go :stuck_out_tongue_winking_eye: a little bit.

**Comment by duglin** [comment]
@vdemeester thanks!  LGTM
ping @calavera for a re-LGTM

**Comment by calavera** [comment]
LGTM

**Pull request merged**